### PR TITLE
More resilient schema lookup

### PR DESCRIPTION
**Pull request by sparrovv**
It happened to me once or twice that specs were running for a long time,
and then they've failed b/c there was a problem with iglucentral
connection.

I don't have the exception at hand, and couldn't reproduce it now, but I believe it was a flaky network connection or maybe iglucentral wasn't available. I've added `read_timeout` and simple retry in case it happens again.

Hopefully this will make it more robust.

### CODE REVIEW

**Comment by chuwy** [comment]
Thanks again, @sparrovv! This also makes it more consistent with canonical Scala Iglu client.

**Comment by snowplowcla** [comment]
@sparrovv has signed the Software Grant and Corporate Contributor License Agreement

**Review by chuwy — APPROVED** [review]

File: lib/iglu-client/registries.rb [hunk 1]
```diff
@@ -41,13 +41,23 @@ def initialize(config, uri)
...
 
-      def lookup_schema(schema_key)
+      def lookup_schema(schema_key, max_retries = 3)
         schema_uri = "#{@uri}/schemas/#{schema_key.as_path}"
+        times_retried = 0
+
         begin
-          response = HTTParty.get(schema_uri)
+          response = HTTParty.get(schema_uri, timeout: 3)
         rescue SocketError => _
```
Comment (sparrovv) [inline lib/iglu-client/registries.rb:50]: I'm just thinking that we might also want to retry on the SocketError 🤔

Comment (chuwy) [reply lib/iglu-client/registries.rb:50]: I don't have a strong opinion on that. But leaning towards not retrying on the `SocketError` as this is rarely something that can be fixed by bigger delay.

**Comment by pvdb** [comment]
Why only catch and retry on `Net::ReadTimeout` but not `Net::ConnectTimeout`?

**Comment by oguzhanunlu** [comment]
Cherry-picked to #14 . Closing.

**Pull request closed without merging**